rumqttc/src/lib.rs

```diff
@@ -404,6 +409,14 @@ impl NetworkOptions {
         self.conn_timeout
     }
 
+    /// bind connection to a specific socket address
+    pub fn set_bind_addr(&mut self, bind_addr: impl ToSocketAddrs) -> &mut Self {
+        self.bind_addr = bind_addr
+            .to_socket_addrs()
+            .map_or(None, |mut iter| iter.next());
```

Comment:
is it fine to just ignore if there was any error in converting to socket address? 

Someone can mistakenly supply invalid addr and think it's working just fine.

for now, as there is no mechanism for returning error here, maybe we can just panic or simply ignore just as we do rn.

which one would you prefer?

Reply:
If it may cause an error, then I think we should not hide the error, either panic or throw. In this case, I think we can simply throw an error since there is no need to support too much chain calls.

```
/// bind connection to a specific socket address
pub fn set_bind_address(&mut self, bind_address: ToSocketAddrs) -> std::io::Result<()> {
    self.bind_address = bind_address.to_socket_addrs()?.next();
    Ok(())
}
```

Reply:
I have also considered this issue and feel that return `Result` is better.

But `set_bind_device` is just like this.

Reply:
And currently v5 is supported, they all use `rumqttc::NetworkOptions` option, but v5 client set it by `MqttOptions`, and client set it by `Eventloop`  just like:
```
    let mut mqtt_options_v5 = rumqttc::v5::MqttOptions::parse_url(url.clone()).unwrap();
    let mut network_options = rumqttc::NetworkOptions::new();
    network_options.set_bind_addr(*crate::app::addrs::LOCAL_ADDR);
    mqtt_options_v5.set_network_options(network_options);
```

Reply:
Since I'm not sure whether to return an error or mask an error, I think we shouldn't wrap the error behavior too much, so I think we should not use ToSocketAddrs, and this conversion should be decided by the caller (because we are not a low-level crate)

Reply:
Yes, it wouldn't be a problem if we used SocketAddr instead of ToSocketAddrs.